File: c++ code/week1/examples/binary image processing/binary image processing/thresholding.cpp
#include <opencv2/opencv.hpp>

int main1() {
	// ead an image in grayscale
	cv::Mat src = cv::imread("../../../../../images/threshold.png", cv::IMREAD_GRAYSCALE);
	cv::Mat dst;

	// set threshold and maximum vlaues
	double thresh = 0;
	double maxValue = 255;

	// binary threshold
	// val > thresh -> val = MaxValue
	// val <= threh -> val = 0
	cv::threshold(src, dst, thresh, maxValue, cv::THRESH_BINARY);

	// display images
	cv::imshow("Original Image", src);
	cv::imshow("Threshold Image", dst);

	cv::waitKey(0);

	return EXIT_SUCCESS;
}

File: python code/binary image processing/thresholding.py
import cv2

# read an image in grayscale
src = cv2.imread("../../images/threshold.png", cv2.IMREAD_GRAYSCALE)

# set threshold and maximum value
thresh = 0
maxValue = 255

# binary threshold
th, dst = cv2.threshold(src=src, thresh=thresh, maxval=maxValue, type=cv2.THRESH_BINARY)

# display images
cv2.imshow("Original Image", src)
cv2.imshow("Thresholded Image", dst)

cv2.waitKey(0)
